### PR TITLE
Add `teleport discovery bootstrap` command

### PR DESCRIPTION
**Pull request by lxea**
Example output:
```
[ec2-user@ip-172-31-33-135 ~]$ sudo ./teleport discovery bootstrap 
Reading configuration at "/etc/teleport.yaml"...

AWS
1. Create IAM Policy "TeleportEC2Discovery":
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "ec2:DescribeInstances",
                "ssm:GetCommandInvocation",
                "ssm:SendCommand"
            ],
            "Resource": [
                "*"
            ]
        }
    ]
}

2. Create IAM Policy "TeleportEC2DiscoveryBoundary":
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "ec2:DescribeInstances",
                "ssm:GetCommandInvocation",
                "ssm:SendCommand"
            ],
            "Resource": [
                "*"
            ]
        }
    ]
}

3. Create SSM Document "TeleportDiscoveryInstaller":

schemaVersion: '2.2'
description: aws:runShellScript
parameters:
  token:
    type: String
    description: "(Required) The Teleport invite token to use when joining the cluster."
  scriptName:
    type: String
    description: "(Required) The Teleport installer script to use when joining the cluster."
mainSteps:
- action: aws:downloadContent
  name: downloadContent
  inputs:
    sourceType: "HTTP"
    destinationPath: "/tmp/installTeleport.sh"
    sourceInfo:
      url: "https://alextest.teleportdemo.net:443/webapi/scripts/{{ scriptName }}"
- action: aws:runShellScript
  name: runShellScript
  inputs:
    timeoutSeconds: '300'
    runCommand:
      - /bin/sh /tmp/installTeleport.sh "{{ token }}"


4. Attach IAM policies to "alex-discovery-role".

Confirm? [y/N]: y
✅[AWS] Create IAM Policy "TeleportEC2Discovery"... done.
✅[AWS] Create IAM Policy "TeleportEC2DiscoveryBoundary"... done.
✅[AWS] Create IAM SSM Document "TeleportDiscoveryInstaller"... done.
✅[AWS] Attach IAM policies to "alex-discovery-role"... done.
```

### CODE REVIEW

**Comment by r0mant** [comment]
@smallinsky @gabrielcorado Can you folks PTAL at this as well? This is configurator for the discovery service similar to what we did for database agent.

**Review by smallinsky — COMMENTED** [review]
First pass. 
@gabrielcorado Could you also take a look ?

File: lib/configurators/aws/awsconfigurator.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/config"
+	"github.com/gravitational/teleport/lib/configurators"
+)
+
+// BuildConfigurators reads the configuration and returns a list of
+// configurators. Configurators that are "empty" are not returned.
+func BuildConfigurators(flags configurators.BootstrapFlags) ([]configurators.Configurator, error) {
```
Comment (gabrielcorado) [inline lib/configurators/aws/awsconfigurator.go:26]: Can we consider moving this function back to the `configurators` package?

This function was intended to provide multiple configurators to the caller. For example, if we add an Azure Configurator, it should return it if it isn't empty. Moving to the `aws` package makes it lose its purpose, only working as a "utility" function to get `config.FileConfig` before initializing the AWS configurator.

File: lib/configurators/aws/aws_test.go [hunk 2]
```diff
@@ -452,6 +458,48 @@ func TestAWSIAMDocuments(t *testing.T) {
 				},
 			},
 		},
+		"AudoDiscovery EC2": {
```
Comment (gabrielcorado) [inline lib/configurators/aws/aws_test.go:461]:
```suggestion
		"AutoDiscovery EC2": {
```

**Review by gabrielcorado — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**